skiplist/src/arena.rs

```diff
@@ -70,3 +71,28 @@ impl Arena {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_arena() {
+        // There is enough space
+        let arena = Arena::with_capacity(128 * 1024);
+        let offset = arena.alloc(8, 8);
+        assert_eq!(offset, 8);
+        assert_eq!(arena.len(), 16);
+
+        unsafe {
+            let ptr = arena.get_mut::<u64>(offset);
+            let offset = arena.offset::<u64>(ptr);
+            assert_eq!(offset, 8);
+        }
+
```

Comment:
Allocate a more aligned ptr on the same arena: `alloc(16,16)`

Reply:
good catch

Reply:
Currently we simply alloc align to 8